### PR TITLE
docs(rust): Fix guide on how to use clippy instead of cargo check on save

### LINKED ISSUE #766: rust analyzer configuration not working using examples from Rust Pack readme
### Checklist

- [X] I have searched through the AstroNvim docs
- [X] I have searched through the existing issues of AstroNvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.9.5

### Operating system/version

Mac 14.2.1

### Terminal/GUI

Iterm2

### Describe the bug

I have this in my init.lua

```lua
  lsp = {
    config = {
      rust_analyzer = {
        settings = {
          -- Add clippy lints for Rust.
          checkOnSave = {
            allFeatures = true,
            command = "clippy",
            extraArgs = { "--no-deps" },
          },
        },
      },
    },
  },
```
Clippy is still not activated and none of these settings seems to be respected

### Steps to Reproduce

Basic setup will reproduce this

### Expected behavior

I expect these settings to work

### Screenshots

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ewoolsey**
Hope this helps someone else!

Closes #766

### CODE REVIEW

**Comment by ewoolsey** [comment]
This fixes my problem
https://github.com/AstroNvim/astrocommunity/issues/766

File: lua/astrocommunity/pack/rust/README.md [hunk 1]
```diff
@@ -60,4 +60,5 @@ If you'd like to run cargo clippy instead of cargo check on safe, add the follow
       },
     },
   },
+
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/rust/README.md:63]: This is unnecessary

**Comment by Uzaaft** [comment]
Pr title needs to be fixed

**Comment by Uzaaft** [comment]
At least as a bare minimum you should read the contribution guide…

File: lua/astrocommunity/pack/rust/README.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ If you'd like to run cargo clippy instead of cargo check on safe, add the follow
     config = {
       rust_analyzer = {
         settings = {
-            -- Add clippy lints for Rust.
+          ["rust-analyzer"] = {
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/rust/README.md:52]:
```suggestion
          -- Add clippy lints for Rust.
          ["rust-analyzer"] = {
```

File: lua/astrocommunity/pack/rust/README.md [hunk 3]
```diff
@@ -60,4 +61,5 @@ If you'd like to run cargo clippy instead of cargo check on safe, add the follow
       },
     },
   },
+
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/rust/README.md:64]:
```suggestion
```

**Review by Uzaaft — APPROVED** [review]

**Pull request merged**